### ISSUE TITLE
[Bug]  I am trying to  port MarianMT using tvm for vulkan. While compiling getting an error assertion error.

### ISSUE THREAD

**Issue by musram** [mlc-ai/mlc-llm#1416]
## 🐛 Bug:
When compiling the model with mlc_chat compile getting the error  r_loops assertion error.

This is happening at   op.gelu(self.fc1(hidden_states))  which i could know as 
gelu: R.Tensor((1, seq_len, 2048), dtype="float32") = R.nn.gelu(add6)
permute_dims10: R.Tensor((2048, 512), dtype="float32") = R.permute_dims(model_encoder_layers_0_fc2_weight, axes=None)
matmul7: R.Tensor((1, seq_len, 512), dtype="float32") = R.matmul(gelu, permute_dims10, out_dtype="void")
            
class EncoderLayer(nn.Module):
    def __init__(self, config: MarianConfig):
        super().__init__()
        self.embed_dim = config.d_model
        self.self_attn = MarianAttention(
            embed_dim=self.embed_dim,
            num_heads=config.encoder_attention_heads,
            kv_cache_len=0,  # no need for kv_cache
            bias=True,
        )
        self.self_attn_layer_norm = nn.LayerNorm(self.embed_dim)
        self.fc1 = nn.Linear(self.embed_dim, config.encoder_ffn_dim, bias=True)
        self.fc2 = nn.Linear(config.encoder_ffn_dim, self.embed_dim, bias=True)
        self.final_layer_norm = nn.LayerNorm(self.embed_dim)

    def forward(self, hidden_states: Tensor, attention_mask: Optional[Tensor] = None) -> Tensor:
        residual = hidden_states
        hidden_states, _ = self.self_attn(
            hidden_states=hidden_states,
            key_value_states=None,
            cached_cross_attn_states=None,
            attention_mask=attention_mask,
            total_seq_len=None,
        )
        hidden_states = residual + hidden_states
        hidden_states = self.self_attn_layer_norm(hidden_states)

        residual = hidden_states
        hidden_states = op.gelu(self.fc1(hidden_states))
        hidden_states = self.fc2(hidden_states)
        hidden_states = residual + hidden_states
        hidden_states = self.final_layer_norm(hidden_states)

        hidden_states = hidden_states.maximum(tir.min_value(hidden_states.dtype)).minimum(
            hidden_states
        )

        return hidden_states

I was able to produce the prim_func that created this issue.
g_var: I.GlobalVar("matmul7")
original func: # from tvm.script import tir as T

@T.prim_func(private=True)
def main(A: T.Buffer((T.int64(1), T.int64(8), T.int64(1), T.int64(1)), "float32"), B: T.Buffer((T.int64(1), T.int64(8), T.int64(1), T.int64(64)), "float32"), matmul: T.Buffer((T.int64(1), T.int64(8), T.int64(1), T.int64(64)), "float32")):
    T.func_attr({"tir.noalias": T.bool(True)})
    # with T.block("root"):
    for i0, i1, i2, i3, k in T.grid(T.int64(1), T.int64(8), T.int64(1), T.int64(64), T.int64(1)):
        with T.block("matmul"):
            v_i0, v_i1, v_i2, v_i3, v_k = T.axis.remap("SSSSR", [i0, i1, i2, i3, k])
            T.reads(A[v_i0, v_i1, v_i2, v_k], B[v_i0, v_i1, v_k, v_i3])
            T.writes(matmul[v_i0, v_i1, v_i2, v_i3])
            with T.init():
                matmul[v_i0, v_i1, v_i2, v_i3] = T.float32(0)
            matmul[v_i0, v_i1, v_i2, v_i3] = matmul[v_i0, v_i1, v_i2, v_i3] + A[v_i0, v_i1, v_i2, v_k] * B[v_i0, v_i1, v_k, v_i3]

## To Reproduce

I am attaching the code to reproduce the above error.
[tvmcs_debug.zip](https://github.com/mlc-ai/mlc-llm/files/13637936/tvmcs_debug.zip)

File "tvm/_ffi/_cython/./packed_func.pxi", line 332, in tvm._ffi._cy3.core.PackedFuncBase.__call__
  File "tvm/_ffi/_cython/./packed_func.pxi", line 263, in tvm._ffi._cy3.core.FuncCall
  File "tvm/_ffi/_cython/./packed_func.pxi", line 252, in tvm._ffi._cy3.core.FuncCall3
  File "tvm/_ffi/_cython/./base.pxi", line 182, in tvm._ffi._cy3.core.CHECK_CALL
  File "tvm/_ffi/_cython/./packed_func.pxi", line 56, in tvm._ffi._cy3.core.tvm_callback
  File "/home/munusairam/softwares/programs/python_projects/tvm-framework/source-build-support/mlc-llm/venv/lib/python3.11/site-packages/tvm/ir/transform.py", line 307, in _pass_func
    return inst.transform_module(mod, ctx)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/munusairam/softwares/programs/python_projects/tvm-framework/source-build-support/mlc-llm/venv/lib/python3.11/site-packages/tvm/dlight/base/transform.py", line 67, in transform_module
    sch = _apply_rules(func, target, self.rules, tunable=False)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/munusairam/softwares/programs/python_projects/tvm-framework/source-build-support/mlc-llm/venv/lib/python3.11/site-packages/tvm/dlight/base/transform.py", line 84, in _apply_rules
    space = rule.apply(func, target, tunable)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/munusairam/softwares/programs/python_projects/tvm-framework/source-build-support/mlc-llm/venv/lib/python3.11/site-packages/tvm/dlight/gpu/gemv.py", line 191, in apply
    is_inner_reduction = normalize(sch, block_info)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/munusairam/softwares/programs/python_projects/tvm-framework/source-build-support/mlc-llm/venv/lib/python3.11/site-packages/tvm/dlight/gpu/gemv.py", line 145, in normalize
    assert r_loops
AssertionError

## Environment

 - Platform (e.g. WebGPU/Vulkan/IOS/Android/CUDA):
 - Operating system (e.g. Ubuntu/Windows/MacOS/...):
 - Device (e.g. iPhone 12 Pro, PC+RTX 3090, ...)
 - How you installed MLC-LLM (`conda`, source):
 - How you installed TVM-Unity (`pip`, source):
 - Python version (e.g. 3.10):
 - GPU driver version (if applicable):
 - CUDA/cuDNN version (if applicable):
 - TVM Unity Hash Tag (`python -c "import tvm; print('\n'.join(f'{k}: {v}' for k, v in tvm.support.libinfo().items()))"`, applicable if you compile models):
 - Any other relevant information:

**Comment by musram** [comment]
Sorry was able to resolve. So will close. I have used op.gelu and should have used nn.gelu.

**Comment by musram** [comment]
I am getting the above assertion r_loop failure error at different matmul. The same code run multiple time gives same error for different matmul. How to fix it?

**Comment by musram** [comment]
[output.txt](https://github.com/mlc-ai/mlc-llm/files/13647449/output.txt)

**Comment by musram** [comment]
Was able to solve.
